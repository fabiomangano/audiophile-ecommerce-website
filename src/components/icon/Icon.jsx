import React from 'react';
import PropTypes from 'prop-types';
import icons from './icons';

export const DIRECTIONS = {
  TOP: 0,
  RIGHT: 90,
  BOTTOM: 180,
  LEFT: -90,
};

export const ICONS = icons;

const Icon = ({ path, color, size, direction }) => {
  return (
    <svg width={size} height={size} viewBox="0 0 24 24">
      <g transform={`rotate(${direction}, 12, 12)`}>
        <path fill={color} d={path} />
      </g>
    </svg>
  );
};

Icon.propTypes = {
  path: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.number,
  direction: PropTypes.number,
};

Icon.defaultProps = {
  size: 24,
  color: 'currentColor',
  direction: DIRECTIONS.TOP,
};

export default Icon;
