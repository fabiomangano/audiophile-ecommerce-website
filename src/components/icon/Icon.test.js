import React from 'react';
import { create } from 'react-test-renderer';
import Icon from './Icon';

describe('Icon', () => {
  it('should renders corretly', () => {
    const tree = create(<Icon path="some/path/to/svg" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
