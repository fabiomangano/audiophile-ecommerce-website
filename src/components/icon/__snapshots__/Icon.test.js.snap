// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should renders corretly 1`] = `
<svg
  height={24}
  viewBox="0 0 24 24"
  width={24}
>
  <g
    transform="rotate(0, 12, 12)"
  >
    <path
      d="some/path/to/svg"
      fill="currentColor"
    />
  </g>
</svg>
`;
